roxie/ccd/ccdprotocol.cpp

```diff
@@ -1601,10 +1598,24 @@ class RoxieSocketWorker : public ProtocolQueryWorker
             return;
         }
         if (httpHelper.queryRequestMlFormat()==MarkupFmt_JSON)
-            queryPT.setown(createPTreeFromJSONString(text, flags, options));
+            queryPT.setown(createPTreeFromJSONString(text, flags, (PTreeReaderOptions) options));
         else
-            queryPT.setown(createPTreeFromXMLString(text, flags, options));
+            queryPT.setown(createPTreeFromXMLString(text, flags, (PTreeReaderOptions) options));
         queryPT.setown(httpHelper.checkAddWrapperForAdaptiveInput(queryPT.getClear(), flags));
+        skipProtocolRoot(queryPT, httpHelper, queryName);
+        if (queryPT->hasProp("_stripWhitespaceFromStoredDataset"))
+        {
+            bool stripTag = queryPT->getPropBool("_stripWhitespaceFromStoredDataset");
+            bool stripFlag = options & ptr_ignoreWhiteSpace;
+            if (stripTag != stripFlag)
+            {
+                if (stripTag)
```

Comment:
A comment here about why you had to parse twice and why there is no way to optimize that away might be handy to avoid future refactorings reintroducing the problem